### ISSUE TITLE
Resize Kernels... when is each appropriate?

### ISSUE THREAD

**Issue by jrnail23** [lovell/sharp#3354]
@lovell, I've seen several issues where you've answered by suggesting the use of a specific kernel for various use cases, but I don't have a very good understanding of what each one does at a high level.

It would be really great if the documentation provided some rough guidance on the various kernels, explaining briefly what the primary use case is for each, and discussing the tradeoffs involved.  I'm thinking really simple, quick  bullet points like "The X kernel is fast, but sacrifices quality.  It's great for line drawings, but not very well suited for rich photos, etc."

BTW, thanks for this really impressive lib!

**Comment by lovell** [comment]
Yes, good idea, this should probably also include sample images demonstrating the effect each has on things like high contrast diagonal edges.

There's currently a PR to add sample images to demonstrate the `fit` option at https://github.com/lovell/sharp/pull/3350 and the `kernel` option should probably take a similar approach.

Happy to accept a PR with a similar update, if you're able.

**Comment by jrnail23** [comment]
@lovell , I would love to be able to contribute a PR here, but I'm afraid the reason I'm asking for it is that I simply don't have this knowledge to begin with. 😄
